.github/workflows/unit-tests.yaml

```diff
@@ -55,79 +59,42 @@ jobs:
-
-    - name: Reset code coverage data
-      run: |
-          rm -rf ${{ github.workspace }}/.coverage/
-          COVERAGE_OUT="${{ github.workspace }}/coverage/coverage-${{ inputs.unit-scenario}}.cov"
-          echo "COVERAGE_OUT=$COVERAGE_OUT" >> $GITHUB_ENV
-
-    - name: Test Go
-      if: ${{ inputs.unit-scenario == 'normal' }}
-      run: |
-        cd ${{ github.workspace }}/src/github.com/snapcore/snapd || exit 1
-        ./run-checks --unit
+          cd cmd/ && make distcheck
+    
+    - name: Set Snapd debug
+      if: ${{ inputs.snapd-debug }}
+      run: echo "SNAPD_DEBUG=1" >> $GITHUB_ENV
 
-    - name: Test Go (SNAPD_DEBUG=1)
-      if: ${{ inputs.unit-scenario == 'snapd_debug' }}
-      run: |
-        cd ${{ github.workspace }}/src/github.com/snapcore/snapd || exit 1
-        SKIP_DIRTY_CHECK=1 SNAPD_DEBUG=1 ./run-checks --unit
+    - name: Set Go test Race
+      if: ${{ inputs.go-test-race }}
+      run: echo "GO_TEST_RACE=1" >> $GITHUB_ENV
 
-    - name: Test Go (withbootassetstesting)
-      if: ${{ inputs.unit-scenario == 'withbootassetstesting' }}
-      run: |
-        cd ${{ github.workspace }}/src/github.com/snapcore/snapd || exit 1
-        SKIP_DIRTY_CHECK=1 GO_BUILD_TAGS=withbootassetstesting ./run-checks --unit
+    - name: Set Skip Coverage
+      if: ${{ inputs.skip-coverage }}
+      run: echo "SKIP_COVERAGE=1" >> $GITHUB_ENV
 
-    - name: Test Go (nosecboot)
-      if: ${{ inputs.unit-scenario == 'nosecboot' }}
-      run: |
-        cd ${{ github.workspace }}/src/github.com/snapcore/snapd || exit 1
-        echo "Dropping github.com/snapcore/secboot"
-        # use govendor remove so that a subsequent govendor sync does not
-        # install secboot again
-        # ${{ github.workspace }}/bin/govendor remove github.com/snapcore/secboot
-        # ${{ github.workspace }}/bin/govendor remove +unused
-        SKIP_DIRTY_CHECK=1 GO_BUILD_TAGS=nosecboot ./run-checks --unit
-
-    - name: Test Go (faultinject)
-      if: ${{ inputs.unit-scenario == 'faultinject' }}
-      run: |
-        cd ${{ github.workspace }}/src/github.com/snapcore/snapd || exit 1
-        SKIP_DIRTY_CHECK=1 GO_BUILD_TAGS=faultinject ./run-checks --unit
-
-    - name: Test Go (-race)
-      if: ${{ inputs.unit-scenario == 'race' }}
-      run: |
-        cd ${{ github.workspace }}/src/github.com/snapcore/snapd || exit 1
-        SKIP_DIRTY_CHECK=1 GO_TEST_RACE=1 SKIP_COVERAGE=1 ./run-checks --unit
-
-    - name: Test Go (snapdusergo)
-      if: ${{ inputs.unit-scenario == 'snapdusergo' }}
+    - name: Test Go
       run: |
-        cd ${{ github.workspace }}/src/github.com/snapcore/snapd || exit 1
-        SKIP_DIRTY_CHECK=1 GO_BUILD_TAGS=snapdusergo ./run-checks --unit
+          ./run-checks --unit
 
     - name: Upload the coverage results
-      if: ${{ inputs.gochannel != 'latest/stable' && inputs.unit-scenario != 'race' }}
+      if: ${{ inputs.gochannel != 'latest/stable' && ! inputs.skip-coverage }}
```

Comment:
Do you know the point of the gochanel part of this comparison? I don't see the relevance myself.

Reply:
I honestly have no idea. It definitely doesn't belong there though. I'll move that to the calling workflow so it would be something like:
```yaml
with:
      skip-coverage: ${{ matrix.gochannel == 'latest/stable' && matrix.test-case.skip-coverage }}
```